src/server/api/routers/rooms.ts

```diff
@@ -64,4 +64,23 @@ export const roomsRouter = createTRPCRouter({
       const [res] = await ctx.db.select().from(rooms).where(eq(rooms.id, input.id)).limit(1);
       return res;
     }),
+  delete: teacherProcedure
+    .input(
+      z.object({
+        id: z.string()
+      }),
+    )
+    .mutation( async ({ ctx, input }) => {
+      const checkClasses = await ctx.db.select().from(classes).where(eq(classes.roomId, input.id));
```

Comment:
checkClasses is fine, but it describes what we are trying to do, rather than what the variable is actually holding. So this might be a slightly better name:

```suggestion
      const classesInRoom = await ctx.db.select().from(classes).where(eq(classes.roomId, input.id));
```